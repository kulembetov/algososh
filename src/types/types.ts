// перечисление для состояний элемента
export enum ElementStates {
  // стандартное состояние
  Default = 'default',
  // изменение
  Changing = 'changing',
  // измененное состояние
  Modified = 'modified',
}

// перечисление для направления сортировки
export enum Direction {
  // по возрастанию
  Ascending = 'ascending',
  // по убыванию
  Descending = 'descending',
}

// перечисление для алгоритмов сортировки
export enum SortAlgorithm {
  // сортировка пузырьком
  bubble = 'пузырек',
  // сортировка выбором
  selectsort = 'выбор',
}

// тип данных элемента данных
export type TDataElement = {
  // значение элемента
  value: string | number;
  // состояние элемента
  state: ElementStates;
  // является ли головным
  isHead?: boolean;
  // является ли хвостовым
  isTail?: boolean;
  // связан ли с другим
  isLinked?: boolean;
  // изменяется ли позиция
  changingPosition?: boolean;
  // новое значение
  newValue?: string | number;
};
