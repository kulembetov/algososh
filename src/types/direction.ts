// перечисление направлений сортировки
export enum Direction {
  // по возрастанию
  Ascending = "ascending",
  // по убыванию
  Descending = "descending",
}
