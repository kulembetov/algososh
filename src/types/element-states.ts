// перечисление состояний элемента
export enum ElementStates {
  // стандартное состояние
  Default = 'default',
  // изменяющееся состояние
  Changing = 'changing',
  // модифицированное состояние
  Modified = 'modified',
}
