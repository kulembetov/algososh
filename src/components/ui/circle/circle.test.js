import renderer from 'react-test-renderer';
import { ElementStates } from '../../../types/element-states';
import { Circle } from './circle';

describe('Тест компонента Button:', () => {
  it('Компонент Button без буквы отрисован корректно', () => {
    const circle = renderer.create(<Circle />).toJSON();
    expect(circle).toMatchSnapshot();
  });
  it('Компонент Button c буквой отрисован корректно', () => {
    const circle = renderer.create(<Circle letter='F' />).toJSON();
    expect(circle).toMatchSnapshot();
  });
  it('Компонент Button с HEAD отрисован корректно', () => {
    const circle = renderer.create(<Circle head='F' />).toJSON();
    expect(circle).toMatchSnapshot();
  });
  it('Компонент Button с react-элементом в HEAD отрисован корректно', () => {
    const circle = renderer.create(<Circle head={<Circle />} />).toJSON();
    expect(circle).toMatchSnapshot();
  });
  it('Компонент Button с TAIL отрисован корректно', () => {
    const circle = renderer.create(<Circle tail='F' />).toJSON();
    expect(circle).toMatchSnapshot();
  });
  it('Компонент Button с react-элементом в TAIL отрисован корректно', () => {
    const circle = renderer.create(<Circle head={<Circle />} />).toJSON();
    expect(circle).toMatchSnapshot();
  });
  it('Компонент Button с index отрисован корректно', () => {
    const circle = renderer.create(<Circle index='5' />).toJSON();
    expect(circle).toMatchSnapshot();
  });
  it('Компонент Button с isSmall === true отрисован корректно', () => {
    const circle = renderer.create(<Circle isSmall={true} />).toJSON();
    expect(circle).toMatchSnapshot();
  });
  it('Компонент Button в состоянии default отрисован корректно', () => {
    const circle = renderer
      .create(<Circle state={ElementStates.Default} />)
      .toJSON();
    expect(circle).toMatchSnapshot();
  });
  it('Компонент Button в состоянии changing отрисован корректно', () => {
    const circle = renderer
      .create(<Circle state={ElementStates.Changing} />)
      .toJSON();
    expect(circle).toMatchSnapshot();
  });
  it('Компонент Button в состоянии modified отрисован корректно', () => {
    const circle = renderer
      .create(<Circle state={ElementStates.Modified} />)
      .toJSON();
    expect(circle).toMatchSnapshot();
  });
});
