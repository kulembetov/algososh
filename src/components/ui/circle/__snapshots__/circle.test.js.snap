// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Тест компонента Button: Компонент Button c буквой отрисован корректно 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      F
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Тест компонента Button: Компонент Button без буквы отрисован корректно 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Тест компонента Button: Компонент Button в состоянии changing отрисован корректно 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   changing"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Тест компонента Button: Компонент Button в состоянии default отрисован корректно 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Тест компонента Button: Компонент Button в состоянии modified отрисован корректно 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   modified"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Тест компонента Button: Компонент Button с HEAD отрисован корректно 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head string"
  >
    F
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Тест компонента Button: Компонент Button с TAIL отрисован корректно 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 string"
  >
    F
  </div>
</div>
`;

exports[`Тест компонента Button: Компонент Button с index отрисован корректно 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  >
    5
  </p>
  <div
    className="text text_type_input text_color_input mt-4 absolute tail60 element"
  />
</div>
`;

exports[`Тест компонента Button: Компонент Button с isSmall === true отрисован корректно 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle  small default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Тест компонента Button: Компонент Button с react-элементом в HEAD отрисован корректно 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        className="circle   default"
      >
        <p
          className="text text_type_circle text_color_input letter"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Тест компонента Button: Компонент Button с react-элементом в TAIL отрисован корректно 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        className="circle   default"
      >
        <p
          className="text text_type_circle text_color_input letter"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;
